hazelcast/src/main/java/com/hazelcast/map/impl/operation/KeyBasedMapOperation.java

```diff
@@ -63,6 +63,22 @@ public final Data getValue() {
         return dataValue;
     }
 
+    protected final Object getValueForFiltering() {
+        if (isObjectInMemoryFormat() || isPostProcessing(recordStore)) {
+            return recordStore.getRecord(dataKey).getValue();
+        } else {
+            return dataValue;
+        }
+    }
+
+    protected final Object getDataValueForPublishing() {
```

Comment:
a short javaDoc can be added here to express difference between filtering and publishing